adapters/sharethrough/butler.go

```diff
@@ -0,0 +1,189 @@
+	headers.Add("Accept", "application/json")
+
+	var extBtlrParams openrtb_ext.ExtImpSharethroughExt
+	if err := json.Unmarshal(imp.Ext, &extBtlrParams); err != nil {
+		return nil, err
+	}
+
+	pKey := extBtlrParams.Bidder.Pkey
+
+	var height, width uint64
+	if len(extBtlrParams.Bidder.IframeSize) >= 2 {
+		height, width = uint64(extBtlrParams.Bidder.IframeSize[0]), uint64(extBtlrParams.Bidder.IframeSize[1])
+	} else {
+		height, width = s.Util.getPlacementSize(imp.Banner.Format)
+	}
+
+	return &adapters.RequestData{
+		Method: "POST",
+		Uri: s.UriHelper.buildUri(StrAdSeverParams{
+			Pkey:               pKey,
+			BidID:              imp.ID,
+			ConsentRequired:    s.Util.gdprApplies(request),
+			ConsentString:      s.Util.gdprConsentString(request),
+			Iframe:             extBtlrParams.Bidder.Iframe,
+			Height:             height,
+			Width:              width,
+			InstantPlayCapable: s.Util.canAutoPlayVideo(request.Device.UA, s.UserAgentParsers),
+		}, request.App),
+		Body:    nil,
+		Headers: headers,
+	}, nil
+}
+
+func (s StrOpenRTBTranslator) responseToOpenRTB(strResp openrtb_ext.ExtImpSharethroughResponse, btlrReq *adapters.RequestData) (*adapters.BidderResponse, []error) {
+	var errs []error
+	bidResponse := adapters.NewBidderResponse()
+
+	bidResponse.Currency = "USD"
+	typedBid := &adapters.TypedBid{BidType: openrtb_ext.BidTypeNative}
+
+	if len(strResp.Creatives) == 0 {
+		errs = append(errs, &errortypes.BadInput{Message: "No creative provided"})
+		return nil, errs
+	}
+	creative := strResp.Creatives[0]
+
+	btlrParams, parseHBUriErr := s.UriHelper.parseUri(btlrReq.Uri)
+	if parseHBUriErr != nil {
+		errs = append(errs, &errortypes.BadInput{Message: parseHBUriErr.Error()})
+		return nil, errs
+	}
+
+	adm, admErr := s.Util.getAdMarkup(strResp, btlrParams)
+	if admErr != nil {
+		errs = append(errs, &errortypes.BadServerResponse{Message: admErr.Error()})
+	}
+
+	bid := &openrtb.Bid{
+		AdID:   strResp.AdServerRequestID,
+		ID:     strResp.BidID,
+		ImpID:  btlrParams.BidID,
+		Price:  creative.CPM,
+		CID:    creative.Metadata.CampaignKey,
+		CrID:   creative.Metadata.CreativeKey,
+		DealID: creative.Metadata.DealID,
+		AdM:    adm,
+		H:      btlrParams.Height,
+		W:      btlrParams.Width,
+	}
+
+	typedBid.Bid = bid
+	bidResponse.Bids = append(bidResponse.Bids, typedBid)
+
+	return bidResponse, errs
+}
+
+func (h StrUriHelper) buildUri(params StrAdSeverParams, app *openrtb.App) string {
+	v := url.Values{}
+	v.Set("placement_key", params.Pkey)
+	v.Set("bidId", params.BidID)
+	v.Set("consent_required", fmt.Sprintf("%t", params.ConsentRequired))
+	v.Set("consent_string", params.ConsentString)
+
+	v.Set("instant_play_capable", fmt.Sprintf("%t", params.InstantPlayCapable))
+	v.Set("stayInIframe", fmt.Sprintf("%t", params.Iframe))
+	v.Set("height", strconv.FormatUint(params.Height, 10))
+	v.Set("width", strconv.FormatUint(params.Width, 10))
+
+	var version string
+
+	if app != nil {
```

Comment:
@mansinahar thanks, now that you mention it, I think I messed up with this one. I was actually trying to get the Prebid-Server version running, do you know a way of doing that by any chance? this is for our own logging purpose, ad server side.

Reply:
@maphe So, I believe there's `/version` endpoint that would return the prebid-server version running but I am not aware of a way to retrieve that information from the request passed on to the adapter. But I will let @hhhjort confirm that.

Reply:
Beside the request, is there a config value or a global constant I could access from the adapter?

Reply:
Are you trying to access the actual version of PBS the adapter is running on, or just the version of your adapter code. Other adapters put a global constant in their adapter code to change the version of the adapter that their adapter reports, so that the bidder can know which version of the adapter code generated the request. Do you need version information beyond what version of your adapter your bidder needs to support?

Reply:
@hhhjort Yes I was hopping I could get the actual version of PBS the adapter is running on. We already have a constant for our own adapter version. If confirmed there's no way to get the PBS version, that's ok the adapter version should be enough. Let me know and I'll clean that up.

Reply:
Unfortunately not. Unless you have a better way of letting the code know what version it is, it doesn't seem to be worth the trouble. Every release would need one extra PR to update the release version of the version constant.

Reply:
Ok, thanks for confirming, I got rid of it. I'll let the team know we will only rely on the adapter version.